libminifi/include/io/tls/TLSSocket.h

```diff
@@ -80,10 +77,13 @@ class TLSContext : public SocketContext {
   int16_t initialize(bool server_method = false);
 
  private:
+  static void deleteContext(SSL_CTX* ptr) { SSL_CTX_free(ptr); }
+
   std::shared_ptr<logging::Logger> logger_;
   std::shared_ptr<Configure> configure_;
   std::shared_ptr<minifi::controllers::SSLContextService> ssl_service_;
-  SSL_CTX *ctx;
+  using Context = std::unique_ptr<SSL_CTX, decltype(&deleteContext)>;
+  Context ctx;
```

Comment:
I would localize the context type alias to the implementation, so that pointer semantics are obvious when looking at the declaration. This is probably very subjective because no guidelines come to my mind to support this.
```suggestion
  std::unique_ptr<SSL_CTX, decltype(&deleteContext)> ctx;
```

Reply:
Adding manually, as I do not want to split the change into two commits. The other suggestion has been applied.

Reply:
Not a big deal, just my opinion/preference. I prefer to use convenience aliases only if they bring a large benefit. This is because in the past I was working with a codebase that really overused type aliases and I had no idea what's what while reading the code.